### PR TITLE
Add support for external evaluation as a new access group rule

### PR DESCRIPTION
**Pull request by Noanan**
The new external evaluation access group rule has been added to the Access backend, this PR add this functionality to Terraform to match the backend. 

Related necessary Cloudflare-go PR: https://github.com/cloudflare/cloudflare-go/pull/882

### CODE REVIEW

**Comment by jacobbednarz** [comment]
thanks! can you please add some test coverage to this so that when the upstream change is merged, we can merge this one soon after?

**Comment by Noanan** [comment]
> thanks! can you please add some test coverage to this so that when the upstream change is merged, we can merge this one soon after?

Thanks for the feedback! Just pushed a new commit with tests, documentation and the changelog entry. 
Please lmk if anything is missing!

File: cloudflare/schema_cloudflare_access_group.go [hunk 1]
```diff
@@ -218,5 +218,21 @@ var AccessGroupOptionSchemaElement = &schema.Resource{
 				},
 			},
 		},
+		"external_evaluation": {
+			Type:     schema.TypeList,
+			Optional: true,
```
Comment (jacobbednarz) [inline cloudflare/schema_cloudflare_access_group.go:223]: just for good measure

```suggestion
			Optional: true,
			MaxItems: 1,
```

**Comment by jacobbednarz** [comment]
i'll give the acceptance tests a run tomorrow first thing. today got away from me with other reviews.

File: website/docs/r/access_group.html.markdown [hunk 2]
```diff
@@ -142,6 +142,18 @@ conditions which can be applied. The conditions are:
     }
   }
   ```
+  * `external_evaluation` - (Optional) Pass a user's identity to an external URL as the `include` condition. 
+  Example:
+
+  ```hcl
+  # ... other configuration
+  include {
+    external {
```
Comment (jacobbednarz) [inline website/docs/r/access_group.html.markdown:151]:
```suggestion
    external_evaluation {
```

File: cloudflare/resource_cloudflare_access_policy_test.go [hunk 3]
```diff
@@ -814,3 +814,56 @@ func testAccessPolicyApprovalGroupConfig(resourceID, zone, accountID string) str
...
+    resource "cloudflare_access_policy" "%[1]s" {
+      application_id = "${cloudflare_access_application.%[1]s.id}"
+      name           = "%[1]s"
+      account_id     = "%[3]s"
+      decision       = "allow"
+      precedence     = "1"
+
+      include {
+        external_evaluation = "example@example.com"
+		external_evaluation {
```
Comment (jacobbednarz) [inline cloudflare/resource_cloudflare_access_policy_test.go:861]:
```suggestion
		external_evaluation {
```

**Comment by jacobbednarz** [comment]
acceptance tests all passing with cloduflare-go shimmed in 

```
TF_ACC=1 go test $(go list ./...) -v -run "^TestAccCloudflareAccessPolicy_" -count 1 -parallel 1 -timeout 120m -parallel 1
?   	github.com/cloudflare/terraform-provider-cloudflare	[no test files]
=== RUN   TestAccCloudflareAccessPolicy_ServiceToken
--- PASS: TestAccCloudflareAccessPolicy_ServiceToken (19.92s)
=== RUN   TestAccCloudflareAccessPolicy_AnyServiceToken
--- PASS: TestAccCloudflareAccessPolicy_AnyServiceToken (19.34s)
=== RUN   TestAccCloudflareAccessPolicy_WithZoneID
--- PASS: TestAccCloudflareAccessPolicy_WithZoneID (36.14s)
=== RUN   TestAccCloudflareAccessPolicy_Group
--- PASS: TestAccCloudflareAccessPolicy_Group (19.20s)
=== RUN   TestAccCloudflareAccessPolicy_MTLS
--- PASS: TestAccCloudflareAccessPolicy_MTLS (17.97s)
=== RUN   TestAccCloudflareAccessPolicy_CommonName
--- PASS: TestAccCloudflareAccessPolicy_CommonName (18.29s)
=== RUN   TestAccCloudflareAccessPolicy_EmailDomain
--- PASS: TestAccCloudflareAccessPolicy_EmailDomain (18.36s)
=== RUN   TestAccCloudflareAccessPolicy_Emails
--- PASS: TestAccCloudflareAccessPolicy_Emails (18.55s)
=== RUN   TestAccCloudflareAccessPolicy_Everyone
--- PASS: TestAccCloudflareAccessPolicy_Everyone (17.71s)
=== RUN   TestAccCloudflareAccessPolicy_IPs
--- PASS: TestAccCloudflareAccessPolicy_IPs (20.74s)
=== RUN   TestAccCloudflareAccessPolicy_AuthMethod
--- PASS: TestAccCloudflareAccessPolicy_AuthMethod (22.17s)
=== RUN   TestAccCloudflareAccessPolicy_Geo
--- PASS: TestAccCloudflareAccessPolicy_Geo (18.13s)
=== RUN   TestAccCloudflareAccessPolicy_Okta
--- PASS: TestAccCloudflareAccessPolicy_Okta (19.44s)
=== RUN   TestAccCloudflareAccessPolicy_PurposeJustification
--- PASS: TestAccCloudflareAccessPolicy_PurposeJustification (18.93s)
=== RUN   TestAccCloudflareAccessPolicy_ApprovalGroup
--- PASS: TestAccCloudflareAccessPolicy_ApprovalGroup (22.91s)
=== RUN   TestAccCloudflareAccessPolicy_ExternalEvaluation
--- PASS: TestAccCloudflareAccessPolicy_ExternalEvaluation (21.01s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/cloudflare	329.160s
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/changelog-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/maintainer-only-file-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/tf-log-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/version	[no test files]
```

just pending the upstream merge now.

**Comment by Noanan** [comment]
> acceptance tests all passing with cloduflare-go shimmed in
> 
> ```
> TF_ACC=1 go test $(go list ./...) -v -run "^TestAccCloudflareAccessPolicy_" -count 1 -parallel 1 -timeout 120m -parallel 1
> ?   	github.com/cloudflare/terraform-provider-cloudflare	[no test files]
> === RUN   TestAccCloudflareAccessPolicy_ServiceToken
> --- PASS: TestAccCloudflareAccessPolicy_ServiceToken (19.92s)
> === RUN   TestAccCloudflareAccessPolicy_AnyServiceToken
> --- PASS: TestAccCloudflareAccessPolicy_AnyServiceToken (19.34s)
> === RUN   TestAccCloudflareAccessPolicy_WithZoneID
> --- PASS: TestAccCloudflareAccessPolicy_WithZoneID (36.14s)
> === RUN   TestAccCloudflareAccessPolicy_Group
> --- PASS: TestAccCloudflareAccessPolicy_Group (19.20s)
> === RUN   TestAccCloudflareAccessPolicy_MTLS
> --- PASS: TestAccCloudflareAccessPolicy_MTLS (17.97s)
> === RUN   TestAccCloudflareAccessPolicy_CommonName
> --- PASS: TestAccCloudflareAccessPolicy_CommonName (18.29s)
> === RUN   TestAccCloudflareAccessPolicy_EmailDomain
> --- PASS: TestAccCloudflareAccessPolicy_EmailDomain (18.36s)
> === RUN   TestAccCloudflareAccessPolicy_Emails
> --- PASS: TestAccCloudflareAccessPolicy_Emails (18.55s)
> === RUN   TestAccCloudflareAccessPolicy_Everyone
> --- PASS: TestAccCloudflareAccessPolicy_Everyone (17.71s)
> === RUN   TestAccCloudflareAccessPolicy_IPs
> --- PASS: TestAccCloudflareAccessPolicy_IPs (20.74s)
> === RUN   TestAccCloudflareAccessPolicy_AuthMethod
> --- PASS: TestAccCloudflareAccessPolicy_AuthMethod (22.17s)
> === RUN   TestAccCloudflareAccessPolicy_Geo
> --- PASS: TestAccCloudflareAccessPolicy_Geo (18.13s)
> === RUN   TestAccCloudflareAccessPolicy_Okta
> --- PASS: TestAccCloudflareAccessPolicy_Okta (19.44s)
> === RUN   TestAccCloudflareAccessPolicy_PurposeJustification
> --- PASS: TestAccCloudflareAccessPolicy_PurposeJustification (18.93s)
> === RUN   TestAccCloudflareAccessPolicy_ApprovalGroup
> --- PASS: TestAccCloudflareAccessPolicy_ApprovalGroup (22.91s)
> === RUN   TestAccCloudflareAccessPolicy_ExternalEvaluation
> --- PASS: TestAccCloudflareAccessPolicy_ExternalEvaluation (21.01s)
> PASS
> ok  	github.com/cloudflare/terraform-provider-cloudflare/cloudflare	329.160s
> ?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/changelog-check	[no test files]
> ?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/maintainer-only-file-check	[no test files]
> ?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/tf-log-check	[no test files]
> ?   	github.com/cloudflare/terraform-provider-cloudflare/version	[no test files]
> ```
> 
> just pending the upstream merge now.

THanks

**Comment by Noanan** [comment]
Oops, sorry for accidently closing the branch. Just wanted to thank you @jacobbednarz for the tests and changing the markdown.

**Pull request merged**